league.js

```diff
@@ -0,0 +1,693 @@
+    _lastUpdated: moment.utc(),
+
+    //--------------------------------------------------------------------------
+    // Whether this is the first time through updating or not.
+    //--------------------------------------------------------------------------
+    _firstRosterUpdate: true,
+
+    //--------------------------------------------------------------------------
+    // Canonicalize the username
+    //--------------------------------------------------------------------------
+    canonicalUsername: function(username) {
+        username = username.split(" ")[0];
+        return username.replace("*", "");
+    },
+
+    //--------------------------------------------------------------------------
+    // Refreshes everything
+    //--------------------------------------------------------------------------
+    'refresh': function() {
+        var self = this;
+        self.refreshRosters(function(err, rosters) {
+            if (err) {
+                console.error("Unable to refresh rosters: " + err);
+                throw new Error(err);
+            } else {
+                console.log("Found " + rosters.length + " teams for " + self.options.name);
+            }
+            self._lastUpdated = moment.utc();
+        });
+        self.refreshCurrentRoundSchedules(function(err, pairings) {
+            if (err) {
+                console.error("Unable to refresh schedule: " + err);
+                throw new Error(err);
+            } else {
+                console.log("Found " + pairings.length + " pairings for " + self.options.name);
+            }
+            self._lastUpdated = moment.utc();
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Refreshes the latest roster information
+    //--------------------------------------------------------------------------
+    'refreshRosters': function(callback) {
+        var query_options = {
+            'min-row': 1,
+            'max-row': 100,
+            'min-col': 1,
+            'max-col': 20,
+            'return-empty': true
+        }
+        var self = this;
+        var firstRosterUpdate = self._firstRosterUpdate;
+        if (self._firstRosterUpdate) {
+            self._firstRosterUpdate = false;
+        }
+        spreadsheets.getRows(
+            self.options.spreadsheet,
+            query_options,
+            function(sheet) {
+                return sheet.title.toLowerCase().indexOf('rosters') != -1;
+            },
+            function(err, rows) {
+                if (err) {
+                    if (err != "Unable to find target worksheet") {
+                        return callback(err, rows);
+                    } else {
+                        return callback(undefined, []);
+                    }
+                }
+                var newTeams = [];
+                rows.forEach(function(row) {
+                    if (
+                        !row['teams'].value ||
```

Comment:
I dont really like that the number of boards is hard coded - this may need to be more flexible. Fine for now but #58 to make this configurable some how.